sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/ViewBuilder.java

```diff
@@ -51,18 +55,104 @@ public ViewBuilder setAggregation(Aggregation aggregation) {
   }
 
   /**
-   * sets {@link LabelsProcessorFactory}.
+   * Specify the attributes processor for this view.
    *
-   * @param labelsProcessorFactory labels processor factory.
+   * <p>Note: This resets all attribute filters, baggage appending and other processing.
+   *
+   * <p>Visible for testing.
+   *
+   * @param processor The pre-processor for measurement attributes.
    * @return this Builder.
    */
-  public ViewBuilder setLabelsProcessorFactory(LabelsProcessorFactory labelsProcessorFactory) {
-    this.labelsProcessorFactory = labelsProcessorFactory;
+  public ViewBuilder setAttributesProcessor(AttributesProcessor processor) {
+    this.processor = processor;
     return this;
   }
 
+  /**
+   * Filters measurement attributes using a given filter.
+   *
+   * <p>Note: This runs after all other attribute processing added so far.
+   *
+   * @param keyFilter filter for key names to include.
+   * @return this Builder.
+   */
+  public ViewBuilder filterAttributes(Predicate<String> keyFilter) {
+    this.processor = this.processor.then(AttributesProcessor.filterByKeyName(keyFilter));
+    return this;
+  }
+
+  /**
+   * Filters measurement attributes using a given regex.
+   *
+   * <p>Note: This runs after all other attribute processing added so far.
+   *
+   * @param keyPattern the regular expression for selecting attributes by key name.
+   * @return this Builder.
+   */
+  public ViewBuilder filterAttributes(Pattern keyPattern) {
+    this.processor =
+        this.processor.then(
+            AttributesProcessor.filterByKeyName(StringPredicates.regex(keyPattern)));
+    return this;
+  }
+
+  /**
+   * Appends a static set of attributes to all measurements.
+   *
+   * <p>Note: This runs after all other attribute processing added so far.
+   *
+   * @param extraAttributes The static attributes to append to measurements.
+   * @return this Builder.
+   */
+  public ViewBuilder appendAttributes(Attributes extraAttributes) {
```

Comment:
I'm somewhat curious about the use-case for this particular feature. Do you have a specific use in mind?

Reply:
Besides our integration test? :)

Marking a canary server with appended attributes.

Reply:
Is that not something that would be on the resource and hence automatically associated with the metrics?